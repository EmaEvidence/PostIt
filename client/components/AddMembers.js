import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';

import addNewMemberAction from '../actions/addNewMemberAction';
import searchUserAction from '../actions/searchUserAction';
import clearStoreAction from '../actions/clearStoreAction';

/**
 * [state description]
 * @type {Object}
 */
class AddMembers extends React.Component {
  /**
   * [constructor description]
   * @method constructor
   * @param  {[type]}    props [description]
   * @return {[type]}          [description]
   */
  constructor(props) {
    super(props);
    this.state = {
      user: '',
      termIsEmpty: true,
      searchTerm: '',
      offset: 0,
      pageCount: ''
    };
    this.addMember = this.addMember.bind(this);
    this.onChange = this.onChange.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
    this.checkIfMember = this.checkIfMember.bind(this);
    this.clear = this.clear.bind(this);
  }
  /**
   * [onChange gets search term]
   * @method onChange
   * @param  {[object]} event [data from the search input]
   * @return {[object]}             [populate the redux store with data]
   */
  onChange(event) {
    if (event.target.value.length !== 0) {
      this.setState({
        termIsEmpty: false,
        searchTerm: event.target.value,
      });
      const offSet = this.state.offset || 0;
      this.props.searchUserAction(event.target.value, offSet, this.props.groupId);

    }
  }
  /**
   * [handlePageClick paginates search result]
   * @method handlePageClick
   * @param  {[object]}        data [click event for page numbers]
   * @return {[object]}             [populate the redux store with data]
   */
  handlePageClick(data) {
    const selected = data.selected;
    const offset = Math.ceil(selected * 5);
    this.props.searchUserAction(this.state.searchTerm, offset, this.props.groupId);
  }

  /**
   * [addMember description]
   * @method addMember
   * @param  {[type]}  userId [description]
   * @param  {[type]}  event  [description]
   * @return  {[type]}  event  [description]
   */
  addMember(userId, event) {
    const groupId = this.props.groupId;
    this.props.addNewMemberAction(groupId, userId);
  }

  checkIfMember(userId, userIds) {
    if (userIds.indexOf(userId) < 0) {
      return true;
    }
  }
  clear() {
    this.setState({
      user: '',
      termIsEmpty: true,
      searchTerm: '',
      offset: 0,
      pageCount: ''
    });
    this.props.clearStoreAction('addMember');
    this.props.clearStoreAction('searchUser');
  }
  /**
   * [render ]
   * @method render
   * @return {[type]} []
   */
  render() {
    const searchResult = this.props.searchResult;
    const userIds = [];
    if (this.props.groups !== undefined) {
      (this.props.groups).forEach((group) => {
        if (group.id === this.props.groupId) {
          (group.Users).map((user) => {
            userIds.push(user.id);
          });
        }
      });
    }
    let resultList;
    if (searchResult.length === 0 || this.state.termIsEmpty === true) {
      resultList = (
        <tr>
          <td>
              No Result, Enter a search term or check what you have typed
          </td>
        </tr>
      );
    } else {
      resultList = (searchResult).map(result =>
        (
          <tr key={result.id}>
            { this.checkIfMember(result.id, userIds)}
            <td> { result.username } </td>
            <td> <input
              type="button"
              className="form-control btn custombutton deep-purple lighten-3"
              value={this.checkIfMember(result.id, userIds) ? 'Add' : 'A Member'}
              onClick={this.addMember.bind(null, result.id)}
              disabled={!this.checkIfMember(result.id, userIds)}
            />
            </td>
          </tr>
      ));
    }
    return (
      <div id="addmembers" className="modal fade reg-form">
        <form className="modal-dialog" onSubmit={this.addMember}>
          <div>
            <h2 className="center"> Add Members </h2>
            <h6 className="center"> { this.props.status }</h6>
          </div>
          <div className="form-group">
            <label htmlFor="search"> Enter your search term </label>
            <input type="search" onChange={this.onChange} />
          </div>
          <div className="result-holder">
            <table className="center add-table">
              <tbody>
                { resultList }
              </tbody>
            </table>
          </div>
          <div className="form-group">
            <table>
              <tbody>
                <tr>
                  <td>
                    <ReactPaginate
                      previousLabel={'previous'}
                      nextLabel={'next'}
                      breakLabel={<a href>...</a>}
                      breakClassName={'break-me'}
                      pageCount={this.props.pageCount}
                      marginPagesDisplayed={2}
                      pageRangeDisplayed={5}
                      onPageChange={this.handlePageClick}
                      containerClassName={'pagination'}
                      subContainerClassName={'pages pagination'}
                      activeClassName={'active'}
                    />
                  </td>
                  <td>
                    <button
                      type="button"
                      onClick={this.clear}
                      className="modal-close btn right deep-purple lighten-4 custombutton"
                    >Cancel</button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </form>
      </div>
    );
  }
}

AddMembers.propTypes = {
  addNewMemberAction: PropTypes.func.isRequired,
  groupId: PropTypes.number.isRequired,
  searchUserAction: PropTypes.func.isRequired,
  status: PropTypes.string.isRequired,
  pageCount: PropTypes.string.isRequired,
  clearStoreAction: PropTypes.func.isRequired
};
/**
 * [mapStateToProps makes the data in the redux store available to this component]
 * @method mapStateToProps
 * @param  {[object]}        state [the entire redux store]
 * @return {[object]}              [the bit made available for this component]
 */
function mapStateToProps(state) {
  return {
    groupId: parseInt((state.setUsersReducer.current_group), 10),
    users: state.setUsersReducer.users,
    status: state.addNewMemberReducer.status,
    searchResult: state.searchUserReducer.searchResult,
    pageCount: state.searchUserReducer.pageCount,
    groups: state.getUserGroupsReducer.groups[0]
  };
}
export default connect(mapStateToProps, { addNewMemberAction, searchUserAction, clearStoreAction })(AddMembers);
