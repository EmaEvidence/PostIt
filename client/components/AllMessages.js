import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

/**
 * [checkIfMessageEmpty description]
 * @type {[type]}
 */
export class AllMessages extends React.Component {
  /**
   * [render description]
   * @method render
   * @return {[type]} [description]
   */
  render() {
    const checkIfMessageEmpty = Object.keys(this.props.messages);
    let Messagelist;
    if (checkIfMessageEmpty.length === 0) {
      Messagelist = (
        <p>
            No Message Yet
        </p>
      );
    } else {
      Messagelist = (this.props.messages).map(message =>
        (
          <span>
            <p
              key={message.id}
            >
              { message.message }
              <br />
              <i className="chip">{message.priority}</i>
              <i className="chip">{message.createdAt}</i><br />
            </p>
            <hr />
          </span>
      ));
    }
    return (
      <div className="col-sm-7 deep-purple lighten-4 messagecard">
        <h2>{ this.props.groupName ? `${this.props.groupName} Messages` : 'Please Select a Group.'} </h2>
        <div>
          { Messagelist }
        </div>
      </div>
    );
  }
}
/**
 * [mapStateToProps description]
 * @method mapStateToProps
 * @param  {[type]}        state [description]
 * @return {[type]}              [description]
 */
function mapStateToProps(state) {
  return {
  //  groupId: state.setCurrentGroupReducer.current_group,
    status: state.setCurrentMessagesReducer.status,
    messages: state.setCurrentMessagesReducer.current_messages,
    groupName: state.setCurrentMessagesReducer.current_group
  };
}

AllMessages.propTypes = {
  messages: PropTypes.array.isRequired,
  groupName: PropTypes.string.isRequired,
};
export default connect(mapStateToProps)(AllMessages);
