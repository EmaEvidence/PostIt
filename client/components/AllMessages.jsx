import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

/**
 * [AllMessages displays Allmessages for a group]
 */
export class AllMessages extends React.Component {
  /**
   * [constructor]
   * @method constructor
   * @param  {object}    props [properties of the Component]
   * @return {void}          []
   */
  constructor(props) {
    super(props);
    this.seenMessage = this.seenMessage.bind(this);
  }
  /**
   * [seenMessage marks a message as seen]
   * @method seenMessage
   * @param  {interger}    messageId [id of the message]
   * @return {void}              []
   */
  seenMessage(messageId) {
    document.getElementById(`message${messageId}`).removeEventListener('click', this.seenMessage());
  }
  /**
   * [render displays the html ]
   * @method render
   * @return {ReactElement} [markup]
   */
  render() {
    const checkIfMessageEmpty = Object.keys(JSON.parse(this.props.messages));
    let Messagelist;
    if (checkIfMessageEmpty.length === 0) {
      Messagelist = (
        <p>
            No Message Yet
        </p>
      );
    } else {
      Messagelist = (JSON.parse(this.props.messages)).map(message =>
        (
          <span>
            <p
              key={message.id}
              onmouseEnter={this.seenMessage.bind(null, message.id)}
              id={`message${message.id}`}
            >
              { message.message }
              <br />
              <i className="chip">{message.priority}</i>
              <i className="chip">{(message.createdAt).split('T')[0]}</i><br />
            </p>
            <hr />
          </span>
      ));
    }
    return (
      <div className="col-sm-7 deep-purple lighten-4 messagecard">
        <h2>{ this.props.groupName ? `${this.props.groupName} Messages` : 'Please Select a Group.'} </h2>
        <div>
          { Messagelist }
        </div>
      </div>
    );
  }
}
/**
 * [mapStateToProps makes the store data available]
 * @method mapStateToProps
 * @param  {object}        state [the store date]
 * @return {object}              [ the data needed by the component]
 */
const mapStateToProps = (state) => {
  return {
  //  groupId: state.setCurrentGroupReducer.current_group,
    status: state.setCurrentMessagesReducer.status,
    messages: JSON.stringify(state.setCurrentMessagesReducer.current_messages),
    groupName: state.setCurrentMessagesReducer.current_group
  };
};

AllMessages.propTypes = {
  messages: PropTypes.string.isRequired,
  groupName: PropTypes.string.isRequired,
};
export default connect(mapStateToProps)(AllMessages);
