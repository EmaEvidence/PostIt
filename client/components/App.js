import React from 'react';
import { Router, Route, Link, IndexRoute, hashHistory, browserHistory } from 'react-router';
import SignUp from './SignUp';
import LogIn from './LogIn';
import NavBar from './NavBar';
import CreateMessage from './CreateMessage';
import CreateGroup from './CreateGroup';
import MessageBoard from './MessageBoard';
import Footer from './Footer';
import img from './img/img.jpg';

class App extends React.Component {
  render() {
    return (
      <div>
        <NavBar />
        <div className="container">
          <img src={img} className="responsive-img" alt="Here" />
        </div>
        <SignUp />
        <LogIn />
        <CreateMessage />
        <CreateGroup />
        <MessageBoard />
        <Footer />
      </div>
    );
  }
}

export default App;
