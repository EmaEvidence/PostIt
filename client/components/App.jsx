import React from 'react';
import Navbar from './NavBar';
import Footer from './Footer';
import Main from './Main';

/**
 * Main App structure
 * @method App
 *
 * @return {ReactElement} markup
 */
const App = () => (
  <div>
    <Navbar />
    <Main />
    <Footer />
  </div>
  );

export default App;
