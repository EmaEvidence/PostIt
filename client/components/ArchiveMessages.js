import React from 'react';
import { connect } from 'react-redux';

/**
 * [archivedMessages description]
 * @type {[type]}
 */
class ArchiveMessages extends React.Component {
  /**
   * [render description]
   * @method render
   * @return {[type]} [description]
   */
  render() {
    const archivedMessages = JSON.parse(this.props.archivedMessages);
    let messageList;
    if (archivedMessages.length === 0) {
      messageList = (
        <p>
              No Message Yet
        </p>
      );
    } else {
      messageList = (archivedMessages).map(message =>
          (
            <p
              key={message.id}
            >
              { message.message }
              <br />
              <i className="chip">{message.priority}</i>
              <i className="chip">{message.createdAt}</i><br />
              <i className="chip">Edit</i>
              <i className="chip">Delete</i>
              <hr />
            </p>
        ));
    }
    return (
      <div id="archivemessages" className="modal fade reg-form">
        <div className="modal-dialog">
          <h2> Archived Messages </h2>
          <div className="messagecard">
            { messageList }
          </div>
          <button
            type="button"
            className="form-control modal-close deep-purple lighten-4 custombutton"
          >Close</button>
        </div>
      </div>
    );
  }
}

ArchiveMessages.propTypes = {
  archivedMessages: React.PropTypes.string.isRequired,
};

/**
 * [mapStateToProps description]
 * @method mapStateToProps
 * @param  {[type]}        state [description]
 * @return {[type]}              [description]
 */
function mapStateToProps(state) {
  let archivedMessages = '';
  if (state.getUserGroupsReducer !== undefined) {
    archivedMessages = JSON.stringify(state.archivedMessagesReducer.archivedMessages);
  }
  return {
    archivedMessages
  };
}
export default connect(mapStateToProps)(ArchiveMessages);
