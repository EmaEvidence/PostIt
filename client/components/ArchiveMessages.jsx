import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

/**
 * ArchivedMessages displays every message a user as seen
 *
 * @type {object}
 */
export class ArchiveMessages extends React.Component {
  /**
   * render
   * @method render
   *
   * @return {ReactElement} markup
   */
  render() {
    const archivedMessages = JSON.parse(this.props.archivedMessages);
    let messageList;
    if (archivedMessages.length === 0) {
      messageList = (
        <p>
              No Message Yet
        </p>
      );
    } else {
      messageList = (archivedMessages).map(message =>
          (
            <p
              key={message.id}
            >
              { message.message }
              <br />
              <i className="chip">{message.priority}</i>
              <i className="chip">{message.createdAt}</i><br />
              <i className="chip">Edit</i>
              <i className="chip">Delete</i><br />
              <i className="chip">Seen by {JSON.stringify(message.views)}</i>
            </p>
        ));
    }
    return (
      <div id="archivemessages" className="modal fade reg-form">
        <div className="modal-dialog">
          <h2>Archived Messages</h2>
          <div className="messagecard">
            { messageList }
          </div>
          <button
            type="button"
            className="form-control modal-close deep-purple lighten-4 custombutton"
          >Close</button>
        </div>
      </div>
    );
  }
}

ArchiveMessages.propTypes = {
  archivedMessages: PropTypes.string.isRequired,
};

/**
 * mapStateToProps makes redux store data available to the Component
 * @method mapStateToProps
 *
 * @param  {object} state store
 *
 * @return {object} data needed by the component
 */
const mapStateToProps = (state) => {
  let archivedMessages = '';
  if (state.groupReducer !== undefined) {
    archivedMessages = JSON.stringify(state.archivedMessageReducer.archivedMessages);
  }
  return {
    archivedMessages
  };
};
export default connect(mapStateToProps)(ArchiveMessages);
