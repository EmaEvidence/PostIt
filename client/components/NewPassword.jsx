import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Input from './Input';
import SignUp from './SignUp';
import LogIn from './LogIn';
import resetPasswordAction from '../actions/resetPasswordAction';

/**
 * [NewPassword Component]
 */
class NewPassword extends React.Component {
  /**
   * [sets the state for the login component]
   * @method constructor
   * @param  {object}    props []
   * @return {void}          []
   */
  constructor(props) {
    super(props);
    this.state = {
      confirmPassword: '',
      password: '',
      status: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * [stores form values in the components state whenever it changes]
   * @method onChange
   * @param  {[object]} event [description]
   * @return {[object]}   [description]
   */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }
  /**
   * [sends the form values to the server]
   * @method onSubmit
   * @param  {[object]} event [description]
   * @return {[objec]}   [description]
   */
  onSubmit(event) {
    event.preventDefault();
    if (this.state.password !== this.state.confirmPassword) {
      this.setState({
        status: 'Password Mismatch'
      });
    } else {
      this.setState({
        status: 'Processing'
      });
      this.props.resetPasswordAction(this.state.password, (this.props.location).search);
      this.setState({
        confirmPassword: '',
        password: ''
      });
    }
  }
  /**
   * [displays the form to login]
   * @method render
   * @return {[jsx]} [description]
   */
  render() {
    return (
      <div id="" className="reg-form row">
        <form onSubmit={this.onSubmit} className="newpassword col s4 offset-s4 valign">
          <div className="modal-header">
            <h3 className="form-header" > Reset Password </h3>
            <p className="center"> Please use the Form Below to Reset Your Password </p>
            <p className="center">{this.props.status ? this.props.status : this.state.status}</p>
          </div>
          <Input
            placeholder={'New Password'}
            value={this.state.password}
            required
            action={this.onChange}
            type={'password'}
            name={'password'}
            class={'form-control'}
          />
          <Input
            placeholder={'Confirm New Password'}
            value={this.state.confirmPasswordpassword}
            required
            action={this.onChange}
            type={'password'}
            name={'confirmPassword'}
            class={'form-control'}
          />
          <div className="form-group">
            <Input
              value={'Reset Password'}
              type={'submit'}
              class={'form-control btn deep-purple lighten-3 custombutton'}
            />
          </div>
        </form>
        <SignUp />
        <LogIn />
      </div>
    );
  }
}

NewPassword.propTypes = {
  status: PropTypes.string.isRequired,
  resetPasswordAction: PropTypes.func.isRequired,
};

/**
 * [mapStateToProps makes the store data available]
 * @method mapStateToProps
 * @param  {object}        state [the store date]
 * @return {object}              [ the data needed by the component]
 */
const mapStateToProps = (state) => {
  return {
    status: state.resetPasswordReducer.status
  };
};

export default connect(mapStateToProps, { resetPasswordAction })(NewPassword);
