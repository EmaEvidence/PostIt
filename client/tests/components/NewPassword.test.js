import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { NewPassword } from '../../components/NewPassword';

describe('NewPassword component should', () => {
  it('renders without crashing', () => {
    const event = {
      target: {
        name: 'password',
        value: 'qwerty123@',
      }
    };
    const wrapper = shallow(<NewPassword />);
    wrapper.instance().onChange(event);
    expect(wrapper.state().password).toEqual('qwerty123@');
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('div').length).toBe(3);
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('Input').length).toBe(2);
    expect(wrapper.find('h3').length).toBe(1);
    expect(wrapper.find('p').length).toBe(1);
    expect(wrapper.find('a').text()).toEqual('<SubmitButton />');
    expect(wrapper.find('h3').text()).toEqual(' Reset Password ');
    expect(wrapper.find('p').first().text())
    .toEqual('Please use the Form Below to Reset Your Password');
    expect(Object.keys(wrapper.props()).length).toBe(3);
  });
});
