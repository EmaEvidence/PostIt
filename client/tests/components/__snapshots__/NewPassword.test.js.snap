// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPassword component should renders without crashing 1`] = `
<div
  className="row"
  id=""
>
  <form
    className="newpassword col s4 offset-s4 valign"
    onSubmit={[Function]}
  >
    <div
      className="modal-header"
    >
      <h3
        className="form-header"
      >
         Reset Password 
      </h3>
      <p
        className="center"
      >
        Please use the Form Below to Reset Your Password
      </p>
    </div>
    <Input
      action={[Function]}
      class="form-control"
      name="password"
      placeholder="New Password"
      required={true}
      type="password"
      value=""
    />
    <Input
      action={[Function]}
      class="form-control"
      name="confirmPassword"
      placeholder="Confirm New Password"
      required={true}
      type="password"
      value=""
    />
    <div
      className="form-group"
    >
      <SubmitButton
        type="submit"
        value="Reset Password"
      />
      <a
        className="right"
        href="/"
      >
        <SubmitButton
          type="button"
          value="Return Home"
        />
      </a>
    </div>
  </form>
</div>
`;
